README.md
<p>
  <a href="https://tailwindcss.com/docs/typography-plugin#gh-light-mode-only" target="_blank">
    <img src="./.github/logo-light.svg" alt="Tailwind CSS Typography" width="450" height="70">
  </a>
  <a href="https://tailwindcss.com/docs/typography-plugin#gh-dark-mode-only" target="_blank">
    <img src="./.github/logo-dark.svg" alt="Tailwind CSS Typography" width="450" height="70">
  </a>
</p>

A plugin that provides a set of `format` classes you can use to add beautiful typographic defaults to any vanilla HTML you don't control, like HTML rendered from Markdown, or pulled from a CMS.

---

## Documentation

For full documentation, visit [tailwindcss.com/docs/typography-plugin](https://tailwindcss.com/docs/typography-plugin).

## Community

For help, discussion about best practices, or any other conversation that would benefit from being searchable:

[Discuss the Tailwind CSS Typography plugin on GitHub](https://github.com/tailwindlabs/tailwindcss/discussions)

For casual chit-chat with others using the framework:

[Join the Tailwind CSS Discord Server](https://tailwindcss.com/discord)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

- Nothing yet!

## [0.5.4] - 2022-07-12

### Fixed

- Update `strong` and `code` color styles to inherit from parent ([#276](https://github.com/tailwindlabs/tailwindcss-typography/pull/276))

## [0.5.3] - 2022-07-07

### Added

- Add styles for `tfoot` elements ([#243](https://github.com/tailwindlabs/tailwindcss-typography/pull/243))
- Add `format-h5` and `format-h6` variants ([#273](https://github.com/tailwindlabs/tailwindcss-typography/pull/273))

### Fixed

- Fix format elements `legacy` mode ([#259](https://github.com/tailwindlabs/tailwindcss-typography/pull/259))
- Allow `lead` class to override element styles ([#260](https://github.com/tailwindlabs/tailwindcss-typography/pull/260))
- Fix generation of `format-headings` variant ([#264](https://github.com/tailwindlabs/tailwindcss-typography/pull/264))
- Fix `figure` spacing ([#267](https://github.com/tailwindlabs/tailwindcss-typography/pull/267))
- Fix child combinator `:where` selectors ([#268](https://github.com/tailwindlabs/tailwindcss-typography/pull/267))
- Fix `format-headings` variant to include `h5` and `h6` elements ([#273](https://github.com/tailwindlabs/tailwindcss-typography/pull/273))

## [0.5.2] - 2022-02-14

### Fixed

- Ensure nested selectors using `&:hover` work ([#246](https://github.com/tailwindlabs/tailwindcss-typography/pull/246))

## [0.5.1] - 2022-01-28

### Removed

- Remove `dist` folder and related dependencies ([#226](https://github.com/tailwindlabs/tailwindcss-typography/pull/226))

### Fixed

- Don't generate invalid CSS when given an array of property values ([#224](https://github.com/tailwindlabs/tailwindcss-typography/pull/224))

## [0.5.0] - 2021-12-09

## [0.5.0-alpha.3] - 2021-11-08

### Changes

- Use `:where` to reduce specificity, making it possible to override format children with utilities ([#203](https://github.com/tailwindlabs/tailwindcss-typography/pull/203))
- Support "undoing" format styles using the `not-format` class on a group of child elements ([#205](https://github.com/tailwindlabs/tailwindcss-typography/pull/205))
- Update color palette for v3 ([#206](https://github.com/tailwindlabs/tailwindcss-typography/pull/206))
- Improve customization API, add alternate grays, add dark mode support, use `::marker` instead of pseudo-elements, and add child element variants (eg. `format-h1:underline`)([#216](https://github.com/tailwindlabs/tailwindcss-typography/pull/216))

## [0.5.0-alpha.2] - 2021-11-08

## [0.5.0-alpha.1] - 2021-11-08

## [0.4.1] - 2021-05-24

### Fixed

- Fix list-style modifier selectors ([#137](https://github.com/tailwindlabs/tailwindcss-typography/pull/137))

## [0.4.0] - 2021-01-15

### Fixed

- Colors without `600` variant, breaks everything (#107)
- Fix empty line in firefox for `pre code` tags (#125)

### Added

- Add support for the `start` and `reversed` attribute on `ol` elements (#110)
- Add support for the `type` on `ul` and `ol` elements (#126)

### Changed (internal)

- Bumped dependencies (#103, #115)
- Cleanup/improve readme (#95)
- Reduce package size (#112)

## [0.3.0] - 2020-11-20

## Changed

- Add support for Tailwind CSS v2.0 and drops support for v1.0 (#79, #82, #87)
- Use `extend` for any overrides, assigning directly to `theme.typography` now completely overrides default configuration
- Consistently use `::before` instead of `:before` ([ba33d77](https://github.com/tailwindlabs/tailwindcss-typography/commit/ba33d77f25ab0c239edd7d425349c86f317061e2))
- Read color values from the user's theme instead of only the default theme

## Added

- Add new `className` option for overriding `format` class (#28)
- Add color modifiers by default like `format-blue` for setting link styles (#92)

## [0.2.0] - 2020-07-15

### Changes

- Switches how variants are generated to use the new component variants API in Tailwind 1.5, dropping support for Tailwind < 1.5

## [0.1.4] - 2020-07-15

### Fixes

- Fixes an issue where the `lead` class was not being applied correctly when used in conjunction with responsive variants

## [0.1.3] - 2020-07-14

### Fixes

- Fixes an issue where the configured monospace font family was not used by default for `pre` and `code` blocks

## [0.1.2] - 2020-07-14

### Fixes

- Fixes an issue in our internal rounding logic that would round values like `20` to `2` 👀

## [0.1.1] - 2020-07-14

### Fixes

- Fixes an issue where the plugin was totally broken and didn't work at all because I deleted some code without even testing my changes 🧠

## [0.1.0] - 2020-07-14

### Added

- Everything!

[unreleased]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.4...HEAD
[0.5.4]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.3...v0.5.4
[0.5.3]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.2...v0.5.3
[0.5.2]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.1...v0.5.2
[0.5.1]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.0...v0.5.1
[0.5.0]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.0-alpha.3...v0.5.0
[0.5.0-alpha.3]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.0-alpha.2...v0.5.0-alpha.3
[0.5.0-alpha.2]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.5.0-alpha.1...v0.5.0-alpha.2
[0.5.0-alpha.1]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.4.1...v0.5.0-alpha.1
[0.4.1]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.4.0...v0.4.1
[0.4.0]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.3.0...v0.4.0
[0.3.0]: https://github.com/tailwindlabs/tailwindcss-typography/compare/v0.2.0...v0.3.0

demo/pages/index.js
import Head from 'next/head'
import MarkdownSample from '../components/MarkdownSample.mdx'
import Navbar from '../components/navbar'
import Footer from '../components/footer'
import { ThemeProvider } from '../theme/theme-context'

export default function Index() {
  return (
    <ThemeProvider>
      <Navbar />
      <div className="antialiased text-gray-900 dark:bg-gray-900 dark:text-white">
        <Head>
          <title>Flowbite Typography</title>
          <meta name="description" content="Flowbite Typography is a plugin that allows you to format inline elements with a single class forked from the official Tailwind CSS Typography plugin."></meta>
          <link rel="canonical" href="https://flowbite.com/plugins/typography/" />
        </Head>
        <div className="px-4 pt-4 pb-8 mx-auto max-w-3xl sm:px-6 sm:py-12 lg:max-w-4xl lg:py-16 lg:px-8 xl:max-w-6xl">
          <article>
            <div className="space-y-8 sm:space-y-12 lg:space-y-16">
              <div className="mx-auto space-y-8 max-w-2xl sm:space-y-0 sm:flex sm:items-center sm:justify-between">
                <div className="flex justify-between items-center">
                  <div className="flex items-center">
                    <div>
                      <h1 className="self-center text-2xl font-semibold whitespace-nowrap dark:text-white">Flowbite Typography</h1>
                      <span className="block">
                        <a href="https://tailwindcss.com/docs/typography-plugin" className="text-sm text-gray-600 dark:text-gray-400 hover:underline">Forked from @tailwindcss/typography</a>
                      </span>
                    </div>
                  </div>
                </div>
                <div className="flex justify-content-between">
                  <a href="https://github.com/themesberg/flowbite-typography" className="inline-flex justify-center items-center py-3 px-5 mr-3 w-full text-sm font-medium text-gray-900 bg-white rounded-lg border border-gray-200 sm:w-auto focus:outline-none hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700">
                    <svg
                      className='mr-2 w-4 h-4'
                      aria-hidden='true'
                      focusable='false'
                      data-prefix='fab'
                      data-icon='github'
                      role='img'
                      xmlns='http://www.w3.org/2000/svg'
                      viewBox='0 0 496 512'>
                      <path
                        fill='currentColor'
                        d='M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z'
                      />
                    </svg> <span className="hidden mr-1 sm:inline">View on</span> GitHub
                  </a>
                  <a href="https://flowbite.com/figma/" className="inline-flex justify-center items-center py-3 px-5 mr-2 w-full text-sm font-medium text-gray-900 bg-white rounded-lg border border-gray-200 sm:w-auto focus:outline-none hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700">
                    <img src="/figma.svg" alt="Figma logo" className="mr-2 w-4 h-4" /> Figma file
                  </a>
                </div>
              </div>
              <div className="mx-auto max-w-2xl format lg:format-lg format-blue dark:format-invert">
                <MarkdownSample />
              </div>
            </div>
          </article>
        </div>
      </div>
      <Footer />
    </ThemeProvider>
  )
}

export const config = {
  unstable_runtimeJS: false,
}
